### PR TITLE
Fixes #37012 - Add a placeholder when selecting booltype host params

### PR DESCRIPTION
**Pull request by girijaasoni**
Adding a new bool type host parameter was throwing a react error when the value wasn't manually selected. As compared to other parameters, we should also allow the user to pass empty value. Hence, the placeholder.

### CODE REVIEW

**Review by stejskalleos — COMMENTED** [review]
While adding or editing the parameter I don't see any errors and it works as expected, however, I don't have any option to select an empty value, nor I don't see any place holder, as the PR suggests. The thing is, do we want to allow empty values for the boolean type? That doesn't make sense, boolean is `true` or `false`, nothing else.

booly

**Comment by girijaasoni** [comment]
> While adding or editing the parameter I don't see any errors and it works as expected, however, I don't have any option to select an empty value, nor I don't see any place holder, as the PR suggests. The thing is, do we want to allow empty values for the boolean type? That doesn't make sense, boolean is `true` or `false`, nothing else.
> 
> booly

We are allowing empty values for global and host group parameters , why shall host parameters be different?  or we can cease to allow empty values there as well. We are even handling empty value cases [here](https://github.com/theforeman/foreman/blob/develop/webpack/assets/javascripts/react_app/components/HostDetails/Tabs/Parameters/ViewTableRow.js#L18), 

The error pops up when we create a param and  don't select a value from the drop down manually, the value is displayed as `true` but is saved as nil.

**Comment by stejskalleos** [comment]
> We are allowing empty values for global and host group parameters, why should host parameters be different? 

:+1: TIL, in that case, I agree with unifying the behavior.

> The error pops up when we create a param

But still, I don't have the option to select an empty value. Plus if I select the boolean type, and click on save, I'm still getting the error.
 Does it work for you with the satellite_theme plugin?

booly_error

**Comment by girijaasoni** [comment]
> > We are allowing empty values for global and host group parameters, why should host parameters be different?
> 
> 👍 TIL, in that case, I agree with unifying the behavior.
> 
> > The error pops up when we create a param
> 
> But still, I don't have the option to select an empty value. Plus if I select the boolean type, and click on save, I'm still getting the error. Does it work for you with the satellite_theme plugin?
> 

That's weird. I have tested it on foreman and it should not happen because in these changes we are adding safety checks for toString() [here](https://github.com/theforeman/foreman/pull/9985/commits/7ac9f5afaa99ad75e50605e802ed7780c0680c23#diff-4086bb88eab1f08b8fbe3532c2b050250142af21b2850e6fa7edca966263287eR17) I don't have the satellite_theme plugin. This is how it looks like for me:

Screenshot from 2024-01-10 18-12-26

**Comment by nofaralfasi** [comment]
> But still, I don't have the option to select an empty value. Plus if I select the boolean type, and click on save, I'm still getting the error. Does it work for you with the satellite_theme plugin?
> 
Are you sure that you applied the right changes? 
I tested it both with and without the satellite_theme plugin, and the only thing that isn't working is when I select the `false` value, it is displayed as empty.

**Comment by girijaasoni** [comment]
> > But still, I don't have the option to select an empty value. Plus if I select the boolean type, and click on save, I'm still getting the error. Does it work for you with the satellite_theme plugin?
> 
> Are you sure that you applied the right changes? I tested it both with and without the satellite_theme plugin, and the only thing that isn't working is when I select the `false` value, it is displayed as empty.

Thanks Nofar,  updated it!

File: webpack/assets/javascripts/react_app/components/HostDetails/Tabs/Parameters/ViewTableRow.js [hunk 1]
```diff
@@ -11,11 +11,13 @@ const getValue = param => {
   if (param['hidden_value?']) {
     return '••••••••';
   }
+  if (!param.value && param.value !== false)
```
Comment (nofaralfasi) [inline webpack/assets/javascripts/react_app/components/HostDetails/Tabs/Parameters/ViewTableRow.js]:
```suggestion
  if (param.value === '')
```

**Comment by nofaralfasi** [comment]
Works well on my end.

File: webpack/assets/javascripts/react_app/components/HostDetails/Tabs/Parameters/ViewTableRow.js (hunk 1 above)
Comment (girijaasoni) [reply webpack/assets/javascripts/react_app/components/HostDetails/Tabs/Parameters/ViewTableRow.js]: `''` values are serialised to `null` , so I updated it as `if (param.value === null)`

**Review by stejskalleos — APPROVED** [review]
:green_apple: **LGTM**

Works as expected, the initial problem was on my instance.

**Comment by stejskalleos** [comment]
Thanks @girijaasoni @nofaralfasi

**Pull request merged**